notifications_utils/request_helper.py

```diff
@@ -75,11 +75,18 @@ def rewrite_response_headers(status, headers, exc_info=None):
             if SPAN_ID_HEADER.lower() not in lower_existing_header_names:
                 headers.append((SPAN_ID_HEADER, str(req.span_id)))
 
-            # Some dynamic scan findings
-            headers.append(("Cross-Origin-Opener-Policy", "same-origin"))
-            headers.append(("Cross-Origin-Embedder-Policy", "require-corp"))
-            headers.append(("Cross-Origin-Resource-Policy", "same-origin"))
-            headers.append(("Cross-Origin-Opener-Policy", "same-origin"))
+            def rewrite_response_headers(status, headers, exc_info=None):
+                lower_existing_header_names = {name.lower() for name, value in headers}
+
+                # Set COOP once (needed for security)
+                if "cross-origin-opener-policy" not in lower_existing_header_names:
+                    headers.append(("Cross-Origin-Opener-Policy", "same-origin"))
+
+                # Ensure `Cross-Origin-Resource-Policy: cross-origin` is set
+                if "cross-origin-resource-policy" not in lower_existing_header_names:
+                    headers.append(("Cross-Origin-Resource-Policy", "cross-origin"))
+
+                return start_response(status, headers, exc_info)
```

Comment:
If I'm reading all of this correctly, it looks like just these two blocks would need to be included in place of the original block of code removed, meaning it should look like this instead:

```suggestion
            # Set COOP once (needed for security)
            if "cross-origin-opener-policy" not in lower_existing_header_names:
                headers.append(("Cross-Origin-Opener-Policy", "same-origin"))

            # Ensure `Cross-Origin-Resource-Policy: cross-origin` is set
            if "cross-origin-resource-policy" not in lower_existing_header_names:
                headers.append(("Cross-Origin-Resource-Policy", "cross-origin"))
```

Otherwise, you're redefining the same function within the existing function and some wonkiness is likely to take place with unintended side effects! 😁